test/LaunchDarkly.EventSource.Tests/EventSourceEndToEndTest.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EmbedIO;
+using EmbedIO.Routing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace LaunchDarkly.EventSource.Tests
+{
+    // For most of the EventSource tests, it's adequate to use a fake HttpMessageHandler that provides
+    // the desired response data without really doing HTTP. But we do want to be sure that our I/O logic
+    // works as intended with a real HTTP client, so this class does some basic EventSource tests
+    // against an embedded HTTP server.
```

Comment:
The embedded server library I'm using here is `EmbedIO`. I've used `WireMock.Net` for similar things in the past, but I've had some issues with that one due to the way it tries to use various Microsoft server frameworks under the covers depending on what platform you're on, whereas `EmbedIO` is a self-contained implementation. I didn't end up having to use it for much, since in the other test classes I think a fake message handler is adequate, but I'll keep it in mind for any future tests like this.